hi-ml/src/health_ml/utils/checkpoint_handler.py

```diff
@@ -119,15 +117,19 @@ def get_local_checkpoints_path_or_download(self) -> Path:
         """
         Get the path to the local weights to use or download them.
         """
-        if self.container.local_weights_path:
-            checkpoint_path = self.container.local_weights_path
-        elif self.container.weights_url:
+        if self.container.src_checkpoint_is_local_file:
+            checkpoint_path = Path(self.container.src_checkpoint)
+        elif self.container.src_checkpoint_is_url:
             download_folder = self.container.checkpoint_folder / MODEL_WEIGHTS_DIR_NAME
             download_folder.mkdir(exist_ok=True, parents=True)
-            checkpoint_path = CheckpointHandler.download_weights(url=self.container.weights_url,
-                                                                 download_folder=download_folder)
+            checkpoint_path = self.download_weights(url=self.container.src_checkpoint, download_folder=download_folder)
+        elif self.container.src_checkpoint_is_aml_run_id:
+            downloader = CheckpointDownloader(
+                run_id=self.container.src_checkpoint, download_dir=self.container.outputs_folder
+            )
+            checkpoint_path = downloader.local_checkpoint_path
         else:
-            raise ValueError("Cannot download weights, neither local_weights_path or weights_url are set")
+            raise ValueError("Cannot download weights, src_checkpoint is not set.")
```

Comment:
The error message is out of sync now - it could be the case that `src_checkpoint` is set but that it doesn't satisfy any of the conditions (is_url, is_local_file, is_aml_run_id)